internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -209,31 +318,38 @@ func untar(log *logger.Logger, version string, archivePath, dataDir string) (str
 				err = closeErr
 			}
 			if err != nil {
-				return "", fmt.Errorf("TarInstaller: error writing to %s: %w", abs, err)
+				return UnpackResult{}, fmt.Errorf("TarInstaller: error writing to %s: %w", abs, err)
 			}
 		case mode.IsDir():
 			log.Debugw("Unpacking directory", "archive", "tar", "file.path", abs)
 			// remove any world permissions from the directory
 			_, err = os.Stat(abs)
 			if errors.Is(err, fs.ErrNotExist) {
 				if err := os.MkdirAll(abs, mode.Perm()&0770); err != nil {
-					return "", errors.New(err, "TarInstaller: creating directory for file "+abs, errors.TypeFilesystem, errors.M(errors.MetaKeyPath, abs))
+					return UnpackResult{}, errors.New(err, "TarInstaller: creating directory for file "+abs, errors.TypeFilesystem, errors.M(errors.MetaKeyPath, abs))
 				}
 			} else if err != nil {
-				return "", errors.New(err, "TarInstaller: stat() directory for file "+abs, errors.TypeFilesystem, errors.M(errors.MetaKeyPath, abs))
+				return UnpackResult{}, errors.New(err, "TarInstaller: stat() directory for file "+abs, errors.TypeFilesystem, errors.M(errors.MetaKeyPath, abs))
 			} else {
 				// set the appropriate permissions
 				err = os.Chmod(abs, mode.Perm()&0o770)
 				if err != nil {
-					return "", errors.New(err, fmt.Sprintf("TarInstaller: setting permissions %O for directory %q", mode.Perm()&0o770, abs), errors.TypeFilesystem, errors.M(errors.MetaKeyPath, abs))
+					return UnpackResult{}, errors.New(err, fmt.Sprintf("TarInstaller: setting permissions %O for directory %q", mode.Perm()&0o770, abs), errors.TypeFilesystem, errors.M(errors.MetaKeyPath, abs))
 				}
 			}
 		default:
-			return "", errors.New(fmt.Sprintf("tar file entry %s contained unsupported file type %v", fileName, mode), errors.TypeFilesystem, errors.M(errors.MetaKeyPath, fileName))
+			return UnpackResult{}, errors.New(fmt.Sprintf("tar file entry %s contained unsupported file type %v", fileName, mode), errors.TypeFilesystem, errors.M(errors.MetaKeyPath, fileName))
 		}
 	}
 
-	return hash, nil
+	return UnpackResult{
+		Hash:          hash,
+		VersionedHome: versionedHome,
+	}, nil
+}
+
+func getFileNamePrefix(archivePath string) string {
+	return strings.TrimSuffix(filepath.Base(archivePath), ".tar.gz") + "/" // omitting `elastic-agent-{version}-{os}-{arch}/` in filename
```

Comment:
Should we use `os.Separator` here instead of the hardcoded `"/"`?

```suggestion
	return strings.TrimSuffix(filepath.Base(archivePath), ".tar.gz") + string(os.PathSeparator) // omitting `elastic-agent-{version}-{os}-{arch}/` in filename
```